src/city/view.c

```diff
@@ -553,6 +553,17 @@ void city_view_foreach_valid_map_tile(map_callback *callback1, map_callback *cal
     }
 }
 
+static int move_grid_offset_by_delta(int grid_offset, int x, int y)
+{
+    int grid_x = grid_offset % GRID_SIZE;
+    int grid_y = grid_offset / GRID_SIZE;
+    if (grid_x + x < 0 || grid_x + x >= GRID_SIZE ||
+        grid_y + y < 0 || grid_y + y >= GRID_SIZE) {
+        return 0;
+    }
+    return grid_offset + map_grid_delta(x, y);
+}
+
 static void do_valid_callback(int x, int y, int grid_offset, map_callback *callback)
 {
     if (map_image_at(grid_offset) >= 6) {
```

Comment:
Shouldn't we check for `grid_offset != 0` as well?

Reply:
In the current usage it makes no difference, as `grid_offset` is guaranteed to be valid.

Reply:
I know, but it saves a call to `map_image_at` and is future-proof, in case we start to return for example `-1` for invalid offsets.